table.cc

```diff
@@ -1262,9 +1262,14 @@ future<std::unordered_map<sstring, table::snapshot_details>> table::get_snapshot
     });
 }
 
-future<> table::flush() {
+future<> table::flush(std::optional<db::replay_position> pos) {
+    if (pos && *pos < _flush_rp) {
+        return make_ready_future<>();
+    }
     auto op = _pending_flushes_phaser.start();
-    return _memtables->request_flush().then([op = std::move(op)] {});
+    return _memtables->request_flush().then([this, op = std::move(op), fp = _highest_rp] {
+        _flush_rp = fp;
```

Comment:
What happens if another flush happens in parallel, and the completions are reordered? Should we merge fp into _flush_rp instead of assigning it?

Reply:
s/.then/co_await/?

Reply:
But you are right. I sort of assumed the above serialized memtable flushes, but should not assume. Should be max(rp, curr)